src/cli/lint/internalLints/upgrade/patches/13.1.1/patchRootTsconfig.ts

```diff
@@ -0,0 +1,115 @@
+import { inspect } from 'util';
+
+import json from '@ast-grep/lang-json';
+import { parse, registerDynamicLanguage } from '@ast-grep/napi';
+import fs from 'fs-extra';
+
+import { log } from '../../../../../../utils/logging.js';
+import type { PatchFunction, PatchReturnType } from '../../index.js';
+
+export const patchRootConfig: PatchFunction = async ({
+  mode,
+}): Promise<PatchReturnType> => {
+  let tsconfigFile: string;
+  try {
+    tsconfigFile = await fs.promises.readFile('tsconfig.json', 'utf8');
+  } catch {
+    return {
+      result: 'skip',
+      reason: 'no root tsconfig.json found',
+    };
+  }
+
+  registerDynamicLanguage({ json });
+  const ast = parse('json', tsconfigFile).root();
```

Comment:
Probably doesn't make a hug difference given our tsconfig files are usually fairly small but should we use [parseAsync instead](https://ast-grep.github.io/guide/api-usage/performance-tip.html#prefer-parseasync-over-parse)?